### PR TITLE
style(grapher): Add return type annotations to selection

### PR DESCRIPTION
**Pull request by danyx23**
This commit gets rid of a helper method to keep more precise type information

### CODE REVIEW

**Comment by danyx23** [comment]
@larsyencken this is the only PR of the return type annotations bunch that has a slight code change (getting rid of a helper function that made types more opaque)

File: grapher/selection/SelectionArray.ts [hunk 1]
```diff
@@ -29,47 +28,52 @@ export class SelectionArray {
...
-    private mapBy(col: keyof Entity, val: keyof Entity) {
+    @computed get entityNameToIdMap(): Map<string, number | undefined> {
         return mapBy(
             this.availableEntities,
-            (v) => v[col],
-            (v) => v[val]
+            (e) => e.entityName,
+            (e) => e.entityId
         )
+        //return this.mapBy(OwidTableSlugs.entityName, OwidTableSlugs.entityId)
```
Comment (danielgavrilov) [inline grapher/selection/SelectionArray.ts:45]: We can remove this comment?

**Review by danielgavrilov — APPROVED** [review]
Looks good! Removing `Selection#mapBy` is 👌

File: clientUtils/Util.ts [hunk 2]
```diff
@@ -924,7 +924,10 @@ export const logMe = (
     return descriptor
 }
 
-export const splitArrayIntoGroupsOfN = (arr: any[], maxPerGroup: number) => {
+export const splitArrayIntoGroupsOfN = (
+    arr: any[],
+    maxPerGroup: number
+): any[] => {
```
Comment (larsyencken) [inline clientUtils/Util.ts:930]: Can this one be easily parameterised too?

```suggestion
export const splitArrayIntoGroupsOfN = <T>(
    arr: T[],
    maxPerGroup: number
): T[][] => {
```

File: grapher/selection/SelectionArray.ts (hunk 1 above)
Comment (larsyencken) [inline grapher/selection/SelectionArray.ts:45]:
```suggestion
```

**Review by larsyencken — APPROVED** [review]
Looks good, minor suggestion

File: clientUtils/Util.ts (hunk 2 above)
Comment (danyx23) [reply clientUtils/Util.ts]: Yes, nice! I'll do another round later of removing all any's but good to take this one down right now :)

**Pull request merged**